Userland/DevTools/UserspaceEmulator/Emulator_syscalls.cpp

```diff
@@ -452,16 +452,19 @@ mode_t Emulator::virt$umask(mode_t mask)
     return syscall(SC_umask, mask);
 }
 
-int Emulator::virt$accept(int sockfd, FlatPtr address, FlatPtr address_length)
+int Emulator::virt$accept4(FlatPtr params_addr)
 {
-    socklen_t host_address_length = 0;
-    mmu().copy_from_vm(&host_address_length, address_length, sizeof(host_address_length));
-    auto host_buffer = ByteBuffer::create_zeroed(host_address_length);
-    int rc = syscall(SC_accept, sockfd, host_buffer.data(), &host_address_length);
+    Syscall::SC_accept4_params params;
+    mmu().copy_from_vm(&params, params_addr, sizeof(params));
+    sockaddr_storage addr = {};
+    socklen_t addrlen;
+    mmu().copy_from_vm(&addrlen, (FlatPtr)params.addrlen, sizeof(socklen_t));
+    VERIFY(addrlen <= sizeof(addr));
+    int rc = accept4(params.sockfd, (sockaddr*)&addr, &addrlen, params.flags);
     if (rc < 0)
         return rc;
```

Comment:
This breaks `errno` preservation as well, since we're not calling the kernel directly anymore.

Reply:
Yeah, I just blindly copied that from `getsockname`. Fixed as well. :)